tests/cpp_unit_tests/test_transformer_tap_regulator.cpp

```diff
@@ -0,0 +1,99 @@
+// SPDX-FileCopyrightText: Contributors to the Power Grid Model project <powergridmodel@lfenergy.org>
+//
+// SPDX-License-Identifier: MPL-2.0
+
+#include <power_grid_model/component/transformer_tap_regulator.hpp>
+
+#include <doctest/doctest.h>
+
+namespace power_grid_model {
+
+TEST_CASE("Test transformer tap regulator") {
+    TransformerTapRegulatorInput const input{.id = 1,
+                                             .regulated_object = 2,
+                                             .control_side = ControlSide::from,
+                                             .u_set = 10.0e3,
+                                             .u_band = 1.0e3,
+                                             .status = true,
+                                             .line_drop_compensation_r = 1.0,
+                                             .line_drop_compensation_x = 2.0};
+
+    double const u_rated{10.0e3};
+
+    TransformerTapRegulator transformer_tap_regulator{input, u_rated};
+
+    SUBCASE("Test energized") {
+        CHECK(transformer_tap_regulator.energized(true));
+        CHECK(transformer_tap_regulator.energized(false));
+    }
+
+    SUBCASE("Test math model type") { CHECK(transformer_tap_regulator.math_model_type() == ComponentType::regulator); }
+
+    SUBCASE("Test control side") { CHECK(transformer_tap_regulator.control_side() == ControlSide::from); }
+
+    SUBCASE("Test status") { CHECK(transformer_tap_regulator.status()); }
+
+    SUBCASE("Test get_output") {
+        TransformerTapRegulatorOutput const output = transformer_tap_regulator.get_output(10);
+        CHECK(output.id == 1);
+        CHECK(output.tap_pos == 10);
+    }
+
+    SUBCASE("Test update") {
+        TransformerTapRegulatorUpdate const update{.id = 1,
+                                                   .u_set = 11.0e3,
+                                                   .u_band = 2.0e3,
+                                                   .status = false,
+                                                   .line_drop_compensation_r = 2.0,
+                                                   .line_drop_compensation_x = 4.0};
```

Comment:
```suggestion
        TransformerTapRegulatorUpdate const update{.id = 1,
                                                   .status = false,
                                                   .u_set = 11.0e3,
                                                   .u_band = 2.0e3,
                                                   .line_drop_compensation_r = 2.0,
                                                   .line_drop_compensation_x = 4.0};
```

Reply:
Thanks, this was also one of the warnings in the CI; fixed it already :)